nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -0,0 +1,124 @@
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.blob.ColdStorageHelper;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.LogCaptureFeature;
+import org.nuxeo.runtime.test.runner.LogFeature;
+import org.nuxeo.runtime.test.runner.LoggerLevel;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class, LogFeature.class, LogCaptureFeature.class })
+@Deploy("org.nuxeo.ecm.core.test.tests:OSGI-INF/test-coldstorage-contrib.xml")
+@Deploy("org.nuxeo.ecm.core.api.tests:OSGI-INF/test-blob-provider-inmemory.xml")
+public class TestColdStorage {
+
+    protected static final String FILE_CONTENT = "foo";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Inject
+    protected LogCaptureFeature.Result logCaptureResult;
+
+    @Test
+    public void shouldMoveBlobDocument() throws IOException {
+        DocumentModel documentModel = session.createDocumentModel("/", "AnyFile", "File");
+        documentModel.setPropertyValue("file:content", (Serializable) Blobs.createBlob(FILE_CONTENT));
+        documentModel = session.createDocument(documentModel);
+        assertFalse(documentModel.hasFacet(COLD_STORAGE));
+
+        // move the blob to cold storage
+        moveContentToColdStorage(session, documentModel.getRef());
+        transactionalFeature.nextTransaction();
+
+        documentModel.refresh();
+        assertTrue(documentModel.hasFacet(COLD_STORAGE));
+
+        // check if the `file:content` contains the thumbnail blob
+        checkBlobContent(documentModel, FILE_CONTENT_PROPERTY, FAKE_THUMBNAIL_CONTENT);
+
+        // check if the `coldstorage:coldContent` contains the original file content
+        checkBlobContent(documentModel, COLD_STORAGE_CONTENT_PROPERTY, FILE_CONTENT);
+    }
+
+    @Test
+    @LogCaptureFeature.FilterOn(loggerClass = ColdStorageHelper.class, logLevel = "DEBUG")
+    @LoggerLevel(klass = ColdStorageHelper.class, level = "DEBUG")
+    public void shouldLogMessageWhenMovingDocumentBlobAlreadyOnColdStorage() {
+        DocumentModel documentModel = session.createDocumentModel("/", "MyFile", "File");
+        documentModel.setPropertyValue("file:content", (Serializable) Blobs.createBlob(FILE_CONTENT));
+        documentModel = session.createDocument(documentModel);
+        assertFalse(documentModel.hasFacet(COLD_STORAGE));
+
+        // move the first time
+        moveContentToColdStorage(session, documentModel.getRef());
+        transactionalFeature.nextTransaction();
```

Comment:
Not needed I believe.

Reply:
as above i removed it, mainly our move method will returns the updated document no need for waiting a next transaction :)